### PR TITLE
D8CORE-000: Kernel Tests For All The Layouts

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW 

# Summary
- Fixes bug where you get a 502 error on empty regions. 

# Needed By (Date)
- 1.0.1 release

# Urgency
- Med/High

# Steps to Test

1. Create a new build of acsf.cardinalsites with the stanford_profile with this branch
2. Edit a full page layout (home) and create a few empty sections with 1, 2, and 3 column templates
3. Save and review the page. You should not have PHP errors.
4. Edit the full page layout again and adjust the settings to center and other alignment options in the emtpy regions
5. Save and review the page without errors
6. Edit the page again and add content to each of the empty sections. 
7. Save and view the page. Assert the configuration options are working.

Full disclosure I have not tested this code and edited it directly on github.

# Affected Projects or Products
- D8CORE

# Associated Issues and/or People
- D8CORE

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Review by jbickar — CHANGES_REQUESTED** [review]
IIRC this 502 was caused by end-user actions, thus this is a bug fix, thus this needs a regression test.

**Comment by sherakama** [comment]
This leads to the question. How does one unit test twig templates? Either that or a very long behat test.

**Comment by sherakama** [comment]
@jbickar Ready for review.

**Review by pookmish — APPROVED** [review]
Confirmed the tests fail with the `8.x-1.x` version of the templates.

**Comment by sherakama** [comment]
Smashing.

**Review by jbickar — COMMENTED** [review]
Tank yew.

**Comment by sherakama** [comment]
Ya'll missed the smashmouth nod in this PR. I am very disappointed.

File: tests/src/Kernel/Layout/ThreeColLayoutTest.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   */
+  public function testThreeColLayoutFullProps() {
+    // Boot twig environment.
+    $twig =  \Drupal::service('twig');
+    $template = drupal_get_path('module', 'jumpstart_ui') . '/templates/layouts/three-column.html.twig';
+    $props = $this->getProps();
+    $this->setRawContent((string) twig_render_template($template, $props));
+    $this->assertText("Somebody once told me php unit is gonna rule me");
+    $this->assertText("This aint the most fun that Ive had");
+    $this->assertText("Im looking for the right class to base my tests off of and I wanna go straight to bed");
```
Comment (sherakama) [inline tests/src/Kernel/Layout/ThreeColLayoutTest.php:46]: https://youtu.be/L_jWHffIx5E?t=34

**Pull request merged**